sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -639,4 +639,169 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
         "expressionStr" -> "SUBSTRING(struct1.a, 0, 1)",
         "reason" -> "generation expression cannot contain non-default collated string type"))
   }
+
+  test("Aggregation of arrays built on collated strings") {
+    val tableName = "test_agg_arr_collated"
+    val simple = Seq(
+      // binary
+      ("utf8_binary", Seq("array('aaa')", "array('AAA')"),
+        Seq((Seq("aaa"), 1), (Seq("AAA"), 1))),
+      ("utf8_binary", Seq("array('aaa', 'bbb')", "array('AAA', 'BBB')"),
+        Seq((Seq("aaa", "bbb"), 1), (Seq("AAA", "BBB"), 1))),
+      ("utf8_binary", Seq("array('aaa')", "array('bbb')", "array('AAA')", "array('BBB')"),
+        Seq((Seq("aaa"), 1), (Seq("bbb"), 1), (Seq("AAA"), 1), (Seq("BBB"), 1))),
```

Comment:
Do we want to cover UTF_BINARY test case here at all? In previous examples we mainly used UTF8_BINARY as validation but the real thing we want to test here are collations other than UTF8_BINARY.